src/ImageSharp/Formats/PixelTypeInfo.cs

```diff
@@ -12,42 +12,37 @@ namespace SixLabors.ImageSharp.Formats;
 /// <summary>
 /// Contains information about the pixels that make up an images visual data.
 /// </summary>
-public class PixelTypeInfo
+public readonly struct PixelTypeInfo
 {
     /// <summary>
-    /// Initializes a new instance of the <see cref="PixelTypeInfo"/> class.
+    /// Initializes a new instance of the <see cref="PixelTypeInfo"/> struct.
     /// </summary>
     /// <param name="bitsPerPixel">Color depth, in number of bits per pixel.</param>
     public PixelTypeInfo(int bitsPerPixel)
         => this.BitsPerPixel = bitsPerPixel;
 
     /// <summary>
-    /// Initializes a new instance of the <see cref="PixelTypeInfo"/> class.
+    /// Gets color depth, in number of bits per pixel.
     /// </summary>
-    /// <param name="bitsPerPixel">Color depth, in number of bits per pixel.</param>
-    /// <param name="alpha">The pixel alpha transparency behavior.</param>
-    public PixelTypeInfo(int bitsPerPixel, PixelAlphaRepresentation alpha)
-    {
-        this.BitsPerPixel = bitsPerPixel;
-        this.AlphaRepresentation = alpha;
-    }
+    public int BitsPerPixel { get; init; }
 
     /// <summary>
-    /// Gets color depth, in number of bits per pixel.
+    /// Gets the count of the color components
     /// </summary>
-    public int BitsPerPixel { get; }
+    public byte ComponentCount { get; init; }
 
     /// <summary>
     /// Gets the pixel alpha transparency behavior.
     /// <see langword="null"/> means unknown, unspecified.
     /// </summary>
-    public PixelAlphaRepresentation? AlphaRepresentation { get; }
-
-    internal static PixelTypeInfo Create<TPixel>()
-        where TPixel : unmanaged, IPixel<TPixel>
-        => new(Unsafe.SizeOf<TPixel>() * 8);
+    public PixelAlphaRepresentation? AlphaRepresentation { get; init; }
 
-    internal static PixelTypeInfo Create<TPixel>(PixelAlphaRepresentation alpha)
+    internal static PixelTypeInfo Create<TPixel>(byte componentCount, PixelAlphaRepresentation pixelAlphaRepresentation)
```

Comment:
I'm wondering whether we can remove this? I want to add the additional enum https://github.com/SixLabors/ImageSharp/discussions/2534#discussioncomment-7171432 which means adding an additional property to the constructor.

Reply:
Maybe just add the additional parameter as having to duplicate `Unsafe.SizeOf<TPixel>() * 8` is worse.

Reply:
@stefannikolei I'm going to add this for you today.

Reply:
Ah. To be honest.. totally forgot about that comment